providers/base/bin/camera_test.py

```diff
@@ -388,51 +389,81 @@ def resolutions(self):
+        if not formats:
+            raise SystemExit("No supported formats found")
+        format = formats[0]
+
+        if self.args.output:
+            self._save_debug_image(format, self.args.device, self.args.output)
+
+        print(
+            "Taking multiple images using the %s format"
+            % format["pixelformat"]
+        )
+
+        for resolution in format["resolutions"]:
+            w = resolution[0]
+            h = resolution[1]
+            f = NamedTemporaryFile(
+                prefix="camera_test_%s%sx%s" % (format["pixelformat"], w, h),
+                suffix=".jpg",
+                delete=False,
             )
-            for res in resolution["resolutions"]:
-                w = res[0]
-                h = res[1]
-                f = NamedTemporaryFile(
-                    prefix="camera_test_%s%sx%s"
-                    % (resolution["pixelformat"], w, h),
-                    suffix=".jpg",
-                    delete=False,
-                )
-                print("Taking a picture at %sx%s" % (w, h))
-                self._still_helper(
-                    f.name, w, h, True, pixelformat=resolution["pixelformat"]
-                )
-                if self._validate_image(f.name, w, h):
-                    print("Validated image %s" % f.name)
-                    os.remove(f.name)
-                else:
-                    print(
-                        "Failed to validate image %s" % f.name, file=sys.stderr
-                    )
-                    os.remove(f.name)
-                    return 1
-            return 0
-
-    def _get_pixel_formats(self, device, maxformats=5):
+            print("Taking a picture at %sx%s" % (w, h))
+            self._still_helper(
+                f.name, w, h, True, pixelformat=format["pixelformat"]
+            )
+            if self._validate_image(f.name, w, h):
+                print("Validated image %s" % f.name)
+                os.remove(f.name)
+            else:
+                print("Failed to validate image %s" % f.name, file=sys.stderr)
+                os.remove(f.name)
+                return 1
+        return 0
+
+    def _save_debug_image(self, format, device, output):
+        """
+        Save an image to a file
+        """
+        # Check if the output directory exists
+        if not os.path.exists(output):
+            raise SystemExit(
+                "Output directory does not exist: {}".format(output)
+            )
+
+        # Choose one resolution image to store as an artifact. We will use
+        # the closest resolution to 640x480 as the target to have some
+        # uniformity in the output
+        target = 640
+        closest_resolution = min(
+            format["resolutions"], key=lambda x: abs(x[0] - target)
+        )
+        w, h = closest_resolution
+        device_name = device.split("/")[-1]
+        filepath = os.path.join(
+            output, "resolution_test_image_{}.jpg".format(device_name)
```

Comment:
Minor: on re-run this is overwritten, maybe we should put a `_{n}.jpg` at the end (you can simply add a len(list(glob(...)))`, also, consider pathlib.

Reply:
After what we decided, this is indeed good enough. If we will ever need the history we will go back to this